### PR TITLE
Migrate 'embedding' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of https://github.com/WordPress/gutenberg/issues/38851.

PR migrates `embedding.test.js` e2e tests to Playwright.

## Why?
See #38851.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/various/embedding.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/various/embedding.spec.js [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+				attributes: {
+					content:
+						'<a href="https://twitter.com/wooyaygutenberg123454312">https://twitter.com/wooyaygutenberg123454312</a>',
+				},
+			},
+		] );
+	} );
+
+	// Reason: A possible regression of https://github.com/WordPress/gutenberg/pull/14705.
+	test.skip( 'should retry embeds that could not be embedded with trailing slashes, without the trailing slashes', async ( {
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/embedding.spec.js:192]: This test is currently failing. When I try to test this scenario manually, it doesn't work at all. I'm skipping the test for now, and we can enable it when regression is fixed.

File: test/e2e/specs/editor/various/embedding.spec.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+			'WordPress valid content. Should render valid figure element.'
+		).toHaveClass( 'wp-block-embed' );
+
+		await embedUtils.insertEmbed(
+			'https://www.youtube.com/watch?v=lXMskKTw3Bc'
+		);
+		await expect(
+			currenEmbedBlock.locator( 'figure' ),
+			'Video content. Should render valid figure element, and include the aspect ratio class.'
+		).toHaveClass( /wp-embed-aspect-16-9/ );
```
Comment (t-hamano) [inline test/e2e/specs/editor/various/embedding.spec.js:147]: If we strictly migrate the original tests, do we also need to check the `.wp-block-embed` and `.is-type-video` classes?

File: test/e2e/specs/editor/various/embedding.spec.js [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+		} );
+
+		const currenEmbedBlock = editor.canvas
+			.getByRole( 'document', { name: 'Block' } )
+			.last();
+
+		await embedUtils.insertEmbed( 'https://twitter.com/notnownikki' );
+		await expect(
+			currenEmbedBlock.locator( 'iframe' ),
+			'Valid embed. Should render valid element.'
```
Comment (t-hamano) [inline test/e2e/specs/editor/various/embedding.spec.js:108]: I didn't know that we could specify an optional message as the second argument 😄

**Review by t-hamano — APPROVED** [review]
LGTM! I tested in debug mode and confirmed that all tests are working as expected.

File: test/e2e/specs/editor/various/embedding.spec.js (hunk 1 above)
Comment (t-hamano) [reply test/e2e/specs/editor/various/embedding.spec.js:192]: I also confirmed that retry doesn't work. Should we open a new issue as a regression?

Comment (Mamaduka) [reply test/e2e/specs/editor/various/embedding.spec.js:192]: Yes, let's open the new issue and reference this PR.

File: test/e2e/specs/editor/various/embedding.spec.js (hunk 2 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/various/embedding.spec.js:147]: I've altered the assertion based on the error message.

Comment (t-hamano) [reply test/e2e/specs/editor/various/embedding.spec.js:147]: I see, that's certainly true 👍

File: test/e2e/specs/editor/various/embedding.spec.js (hunk 1 above)
Comment (t-hamano) [reply test/e2e/specs/editor/various/embedding.spec.js:192]: Submitted: #58002

Comment (Mamaduka) [reply test/e2e/specs/editor/various/embedding.spec.js:192]: Thank you, @t-hamano 🙏

**Pull request merged**